lib/ansible/module_utils/dimensiondata.py

```diff
@@ -0,0 +1,393 @@
+#   - Mark Maglana   <mmaglana@gmail.com>
+#
+# Common methods to be used by versious module components
+import os
+import ConfigParser
+from os.path import expanduser
+from uuid import UUID
+
+try:
+    from libcloud.common.dimensiondata import \
+        API_ENDPOINTS, DimensionDataAPIException
+    HAS_LIBCLOUD = True
+except ImportError:
+    HAS_LIBCLOUD = False
+
+
+# Custom Exceptions
+
+class LibcloudNotFound(Exception):
+    pass
+
+
+class MissingCredentialsError(Exception):
+    pass
+
+
+class UnknownNetworkError(Exception):
+    pass
+
+
+class UnknownVLANError(Exception):
+    pass
+
+
+def check_libcloud_or_fail():
+    """
+    Checks if libcloud is installed and fails if not
+    """
+    if not HAS_LIBCLOUD:
+        raise LibcloudNotFound("apache-libcloud is required.")
+
+
+def get_credentials():
+    """
+    This method will get user_id and key from environemnt or dot file.
+    Environment takes priority.
+
+    To set in environment:
+
+        export DIDATA_USER='myusername'
+        export DIDATA_PASSWORD='mypassword'
+
+    To set in dot file place a file at ~/.dimensiondata with
+    the following contents:
+
+        [dimensiondatacloud]
+        DIDATA_USER: myusername
+        DIDATA_PASSWORD: mypassword
+    """
+    check_libcloud_or_fail()
+
+    # Attempt to grab from environment
+    user_id = os.environ.get('DIDATA_USER', None)
+    key = os.environ.get('DIDATA_PASSWORD', None)
+
+    # Environment failed try dot file
+    if not user_id or not key:
+        home = expanduser('~')
+        config = ConfigParser.RawConfigParser()
+        config.read("%s/.dimensiondata" % home)
+        try:
+            user_id = config.get("dimensiondatacloud", "DIDATA_USER")
+            key = config.get("dimensiondatacloud", "DIDATA_PASSWORD")
+        except ConfigParser.NoSectionError, ConfigParser.NoOptionError:
+            pass
+
+    # One or more credentials not found. Function can't recover from this
+    # so it has to raise an error instead of fail silently.
+    if not user_id:
+        raise MissingCredentialsError("Dimension Data user id not found")
+    elif not key:
+        raise MissingCredentialsError("Dimension Data key not found")
+
+    # Both found, return data
+    return dict(user_id=user_id, key=key)
+
+
+def get_dd_regions():
+    """
+    Get the list of available regions whose vendor is Dimension Data.
+    """
+    check_libcloud_or_fail()
+
+    # Get endpoints
+    all_regions = API_ENDPOINTS.keys()
+
+    # filter to only DimensionData endpoints
+    region_list = filter(lambda i: i.startswith('dd-'), all_regions)
+
+    # Strip prefix
+    regions = [region[3:] for region in region_list]
+
+    return regions
+
+
+def get_network_domain_by_name(driver, name, location):
+    """
+    Get a network domain object by its name
+    """
+    networks = driver.ex_list_network_domains(location=location)
+    found_networks = filter(lambda x: x.name == name, networks)
```

Comment:
This usage of filter won't work on python3.  filter on python3 will return a generator which will evaluate to true in the conditional below.  Use a list comprehension so that found_networks ends up with an empty list on python3 which will evaluate to false.